compileopts/target.go

```diff
@@ -121,15 +121,18 @@ func (spec *TargetSpec) load(r io.Reader) error {
 
 // loadFromGivenStr loads the TargetSpec from the given string that could be:
 // - targets/ directory inside the compiler sources
-// - a relative or absolute path to custom (project specific) target specification .json file;
+// - a relative or absolute path to custom (project specific) target specification .json or .yaml file;
 //   the Inherits[] could contain the files from target folder (ex. stm32f4disco)
-//   as well as path to custom files (ex. myAwesomeProject.json)
+//   as well as path to custom files (ex. myAwesomeProject.yaml)
 func (spec *TargetSpec) loadFromGivenStr(str string) error {
 	path := ""
-	if strings.HasSuffix(str, ".json") {
+	if strings.HasSuffix(str, ".json") || strings.HasSuffix(str, ".yaml") {
 		path, _ = filepath.Abs(str)
 	} else {
-		path = filepath.Join(goenv.Get("TINYGOROOT"), "targets", strings.ToLower(str)+".json")
+		path = filepath.Join(goenv.Get("TINYGOROOT"), "targets", strings.ToLower(str)+".yaml")
+		if _, err := os.Stat(path); errors.Is(err, os.ErrNotExist) {
+			path = filepath.Join(goenv.Get("TINYGOROOT"), "targets", strings.ToLower(str)+".json")
+		}
```

Comment:
What about renaming all files to `.yaml`? YAML is backwards compatible so you could keep the existing JSON files but with a different file extension. Then, in a separate PR, we could mass-format the target JSON/YAML files in a consistent way.